Web3/Classes/ContractABI/Contract/SolidityFunction.swift

```diff
@@ -73,6 +73,8 @@ public protocol SolidityFunction: class {
     /// - Parameter inputs: Input values. Must be in the correct order.
     /// - Returns: Invocation object
     func invoke(_ inputs: ABIEncodable...) -> SolidityInvocation
+
+    func invoke(_ inputs: [ABIEncodable]) -> SolidityInvocation
```

Comment:
This will work for now because Swift 4.1 does not properly understand conditional conformance, but really [ABIEncodable] also conforms to ABIEncodable, and down the road the compiler won't be able to tell the difference between these 2 methods. Might be best to change the signature to distinguish them.

Reply:
(Same goes for all of the other `invoke` methods)

Reply:
Might be a good idea changing this to `func invoke(_ arrayOfInputs: [ABIEncodable]) -> SolidityInvocation)`, what do you say @pixelmatrix?

Reply:
I like the distinction, but to really help the compiler distinguish between the two we'll need one of the methods to not use the underscore. Given that the variadic approach is intended for syntactic sugar, it probably makes a little more sense for the array method to have a label. 

How about:

```swift
func invoke(with inputs: [ABIEncodable]) -> SolidityInvocation {}
```

Which would look something like…

```swift
MyContract['balanceOf']?.invoke(with: [myAddress])
```